File: proj.android_studio/ee-x-core/src/main/java/com/ee/Logger.kt
package com.ee

import android.util.Log

/**
 * Created by enrevol on 4/13/16.
 */
class Logger @JvmOverloads constructor(
    private val _tag: String,
    private var _logLevel: Int = Log.VERBOSE) : ILogger {
    var logLevel: Int
        get() = _logLevel
        set(value) {
            _logLevel = value
        }

    override fun error(message: String) {
        if (_logLevel <= Log.ERROR) {
            Log.e(_tag, message)
        }
    }

    override fun error(message: String, th: Throwable) {
        if (_logLevel <= Log.ERROR) {
            Log.e(_tag, message, th)
        }
    }

    override fun warn(message: String) {
        if (_logLevel <= Log.WARN) {
            Log.w(_tag, message)
        }
    }

    override fun debug(message: String) {
        if (_logLevel <= Log.DEBUG) {
            Log.d(_tag, message)
        }
    }

    override fun info(message: String) {
        if (_logLevel <= Log.INFO) {
            Log.i(_tag, message)
        }
    }
}

File: src/ts/src/core/Logger.ts
import { ILogger } from "./ILogger";
import { LogLevel } from "./LogLevel";

export class Logger implements ILogger {
    private readonly _tag: string;
    private _logLevel: LogLevel;

    public constructor(tag: string) {
        this._tag = tag;
        this._logLevel = LogLevel.Verbose;
    }

    public get logLevel(): LogLevel {
        return this._logLevel;
    }

    public set logLevel(value: LogLevel) {
        this._logLevel = value;
    }

    public error(message: string): void {
        if (this._logLevel <= LogLevel.Error) {
            console.error(`${this._tag}: ${message}`);
        }
    }

    public warn(message: string): void {
        if (this._logLevel <= LogLevel.Warning) {
            console.warn(`${this._tag}: ${message}`);
        }
    }

    public info(message: string): void {
        if (this._logLevel <= LogLevel.Info) {
            console.info(`${this._tag}: ${message}`);
        }
    }

    public debug(message: string): void {
        if (this._logLevel <= LogLevel.Debug) {
            console.debug(`${this._tag}: ${message}`);
        }
    }
}